src/core/Akka.Streams/Implementation/StreamLayout.cs

```diff
@@ -1004,20 +1009,21 @@ private object ResolveMaterialized(StreamLayout.IMaterializedValueNode node, IDi
             object result;
             if (node is StreamLayout.Atomic)
             {
-                if (!values.TryGetValue(((StreamLayout.Atomic)node).Module, out result)) result = null;
+                var atomic = (StreamLayout.Atomic) node;
+                result = values[atomic.Module];
             }
             else if (node is StreamLayout.Combine)
             {
-                var combine = node as StreamLayout.Combine;
+                var combine = (StreamLayout.Combine) node;
                 result = combine.Combinator(ResolveMaterialized(combine.Left, values, indent + "  "),
                     ResolveMaterialized(combine.Right, values, indent + "  "));
             }
             else if (node is StreamLayout.Transform)
             {
-                var transform = node as StreamLayout.Transform;
+                var transform = (StreamLayout.Transform) node;
                 result = transform.Transformator(ResolveMaterialized(transform.Node, values, indent + "  "));
             }
-            else result = null;
+            else result = Unit.Instance;
```

Comment:
Are you sure about this one? Setting up Unit is not the same as not setting any value at all.